central/sensorupgradeconfig/datastore/datastore_impl.go

```diff
@@ -51,6 +53,10 @@ func (d *dataStore) UpsertSensorUpgradeConfig(ctx context.Context, sensorUpgrade
 		return sac.ErrResourceAccessDenied
 	}
 
+	if env.ManagedCentral.BooleanSetting() && sensorUpgradeConfig.EnableAutoUpgrade {
```

Comment:
```suggestion
	if env.ManagedCentral.BooleanSetting() && sensorUpgradeConfig.GetEnableAutoUpgrade() {
```
always prefer getters in protobufs for `nil`-safety